controllers/nodemaintenance_controller.go

```diff
@@ -192,7 +195,12 @@ func (r *NodeMaintenanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}
 	r.logger.Info("All pods evicted", "nodeName", nodeName)
 
+	// Calculate last update time when the drian is over - there are no more pending pods
+	if instance.Status.PendingPods != nil {
```

Comment:
- why this if? Any reason to not set last update here always? 🤔 
- comment does not fit

Reply:
After evicting all the pods we set the _Phase_, _DrainProgress_, _PendingPods_ and _LastUpdate_. Then we update the status, and since there was a change in the status  we run another reconcile. Now there won't be pods to evict and if always set LastUpdate then we will end up with an endless loop of updating the LastUpdate, without ever reaching the end of the reconcile.

IIRC this what happened when I tried it before, but I will give another look.

Reply:
ah, that would make sense. I forgot that our own changes also trigger reconcile, but I think you are right 👍🏼